gui/src/components/onboarding/pages/CalibrationTutorial.tsx

```diff
@@ -32,42 +34,62 @@ export function CalibrationTutorialPage() {
     CalibrationStatus.WAITING
   );
   const [skipButton, setSkipButton] = useState(false);
+  const [settled, setSettled] = useState(false);
   const { timer, isCounting, startCountdown, abortCountdown } = useCountdown({
-    duration: IMU_CALIBRATION_TIME,
-    onCountdownEnd: () => setCalibrationStatus(CalibrationStatus.SUCCESS),
+    duration: settled ? IMU_CALIBRATION_TIME : IMU_SETTLE_TIME,
+    onCountdownEnd: () =>
+      settled
+        ? setCalibrationStatus(CalibrationStatus.SUCCESS)
+        : setSettled(true),
   });
   useTimeout(() => setSkipButton(true), 10000);
   const connectedIMUTrackers = useAtomValue(connectedIMUTrackersAtom);
   const restCalibrationTrackers =
     useRestCalibrationTrackers(connectedIMUTrackers);
   const [rested, setRested] = useState(false);
-  const lastValueMap = useRef(new Map<number, Vector3[]>());
+  const lastValueMap = useRef(new Map<number, Vector3>());
   useEffect(() => {
     const accelLength = restCalibrationTrackers.every((x) => {
       if (
+        x.device?.id?.id === undefined ||
         x.tracker.trackerId?.trackerNum === undefined ||
         x.tracker.trackerId.deviceId?.id === undefined ||
         !x.tracker.linearAcceleration
       )
         return false;
 
-      const trackerId =
-        x.tracker.trackerId.trackerNum + (x.tracker.trackerId.trackerNum << 8);
-      const lastValues = lastValueMap.current.get(trackerId) ?? [];
-      lastValueMap.current.set(trackerId, lastValues);
+      const trackerId = x.device.id.id;
+      //        x.tracker.trackerId.trackerNum + (x.tracker.trackerId.deviceId.id << 8);
+      const lastValue = lastValueMap.current.get(trackerId) ?? new Vector3();
+      lastValueMap.current.set(trackerId, lastValue);
 
       const vec3 = Vector3FromVec3fT(x.tracker.linearAcceleration);
-      if (lastValues.length > 5) {
-        lastValues.shift();
-        const avg = averageVector(lastValues).lengthSq();
-        lastValues.push(vec3);
-        return vec3.lengthSq() <= avg + ACCEL_TOLERANCE ** 2;
+
+      if (vec3.lengthSq() > ACCEL_TOLERANCE ** 2) {
+        return false;
+      }
+
+      const delta = new Vector3();
+      delta.subVectors(lastValue, vec3);
+
+      if (delta.lengthSq() > ACCEL_HYSTERESIS ** 2) {
+        lastValue.copy(vec3);
+        return false;
       }
-      lastValues.push(vec3);
-      return false;
+
+      return true;
     });
 
-    setRested(accelLength || restCalibrationTrackers.length === 0);
+    if (accelLength && !settled && !isCounting) {
+      abortCountdown();
+      startCountdown();
+    } else if (!accelLength && !settled && isCounting) {
+      abortCountdown();
+    } else if (!accelLength && settled) {
+      setSettled(false);
+    }
+
+    setRested(settled || restCalibrationTrackers.length === 0);
   }, [restCalibrationTrackers]);
```

Comment:
```suggestion
  }, [restCalibrationTrackers, settled, isCounting]);
```
need to add it as a dependency